File: images/node/hello_world.js
#!/usr/bin/env node
/**
 * Simple Hello World program in Node.js
 */

const readline = require('readline');

function main() {
    console.log("Hello, World!");
    console.log("Welcome to Node.js!");

    // Create readline interface
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    // Get user input
    rl.question("What's your name? ", (name) => {
        console.log(`Nice to meet you, ${name}!`);
        rl.close();
    });
}

main();

File: images/go/hello_world.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

// Simple Hello World program in Go
func main() {
	fmt.Println("Hello, World!")
	fmt.Println("Welcome to Go!")

	// Get user input
	fmt.Print("What's your name? ")
	reader := bufio.NewReader(os.Stdin)
	name, _ := reader.ReadString('\n')
	name = strings.TrimSpace(name)
	fmt.Printf("Nice to meet you, %s!\n", name)
}
